pkg/testcase/clusterrestore.go

```diff
@@ -0,0 +1,416 @@
+	shared.LogLevel("info", "nodes post restore have been validated")
+
+	testValidatePodsPostRestore()
+	shared.LogLevel("info", "pods post restore have been validated")
+}
+
+func testIngressPostRestore(newServerIP string, applyWorkload, deleteWorkload bool, kubectlCmd string) {
+	if applyWorkload {
+		workloadErr := shared.ManageWorkload("apply", "ingress.yaml")
+		Expect(workloadErr).NotTo(HaveOccurred(), "ingress manifest not deployed")
+	}
+
+	ingressErr := assert.ValidateOnNode(newServerIP, kubectlCmd+" get pods -n test-ingress -l k8s-app=nginx-app-ingress"+
+		" --field-selector=status.phase=Running", "Running")
+	Expect(ingressErr).NotTo(HaveOccurred())
+
+	if deleteWorkload {
+		workloadErr := shared.ManageWorkload("delete", "ingress.yaml")
+		Expect(workloadErr).NotTo(HaveOccurred(), "Ingress manifest not deleted")
+	}
+	shared.LogLevel("info", "ingress successfully validated post cluster restore")
+}
+
+func testNodePortPostRestore(newServerIP string, applyWorkload, deleteWorkload bool, kubectlCmd string) {
+	if applyWorkload {
+		workloadErr := shared.ManageWorkload("apply", "nodeport.yaml")
+		Expect(workloadErr).NotTo(HaveOccurred(), "NodePort manifest not deployed")
+	}
+
+	nodePortErr := assert.ValidateOnNode(newServerIP, kubectlCmd+" get pods -n test-nodeport -l k8s-app=nginx-app-nodeport "+
+		"--field-selector=status.phase=Running", "Running")
+	Expect(nodePortErr).NotTo(HaveOccurred())
+
+	if deleteWorkload {
+		workloadErr := shared.ManageWorkload("delete", "nodeport.yaml")
+		Expect(workloadErr).NotTo(HaveOccurred(), "NodePort manifest not deleted")
+	}
+	shared.LogLevel("info", "nodeport successfully validated post cluster restore")
+}
+
+func testClusterIPPostRestore(newServerIP string, applyWorkload, deleteWorkload bool, kubectlCmd string) {
+	if applyWorkload {
+		workloadErr := shared.ManageWorkload("apply", "clusterip.yaml")
+		Expect(workloadErr).NotTo(HaveOccurred(), "Cluster IP manifest not deployed")
+	}
+
+	clusterIPErr := assert.ValidateOnNode(newServerIP, kubectlCmd+" get pods -n test-clusterip -l"+
+		" k8s-app=nginx-app-clusterip "+"--field-selector=status.phase=Running", "Running")
+	Expect(clusterIPErr).NotTo(HaveOccurred())
+
+	if deleteWorkload {
+		workloadErr := shared.ManageWorkload("delete", "clusterip.yaml")
+		Expect(workloadErr).NotTo(HaveOccurred(), "Cluster IP manifest not deleted")
+	}
+	shared.LogLevel("info", "clusterIP successfully validated post cluster restore")
+}
+
+func testDNSAccessPostRestore(newServerIP, kubectlCmd string) {
+	dnsErr := assert.ValidateOnNode(newServerIP, kubectlCmd+" get pods -n dnsutils dnsutils")
+	Expect(dnsErr).To(HaveOccurred())
+
+	shared.LogLevel("info", "dns successfully validated post cluster restore")
+}
+
+func testValidateNodesPostRestore(newServerIP string) {
+	res, err := shared.GetNodes(true)
+	Expect(err).NotTo(HaveOccurred())
+	fmt.Println("Nodes: ", res)
+	Expect(res).NotTo(BeEmpty())
+	for _, node := range res {
```

Comment:
also tbh if u are looking for something especifc just already go over that, instead of the whole response